lms/djangoapps/support/static/support/jsx/single_support_form.jsx

```diff
@@ -14,140 +14,159 @@ import LoggedInUser from './logged_in_user';
 import LoggedOutUser from './logged_out_user';
 import Success from './success';
 
+const initialFormErrors = {
+  course: undefined,
+  subject: undefined,
+  message: undefined,
+  request: undefined,
+};
 
 class RenderForm extends React.Component {
   constructor(props) {
     super(props);
+    this.submitFormUrl = this.props.context.submitFormUrl;
     this.userInformation = this.props.context.user;
     const course = this.userInformation ? this.userInformation.course_id : '';
+    this.courseDiscussionURL = '/courses/{course_id}/discussion/forum';
     this.state = {
       currentRequest: null,
-      errorList: [],
+      errorList: initialFormErrors,
       success: false,
       formData: {
+        course,
         subject: '',
         message: '',
-        course,
       },
     };
+    this.formValidationErrors = {
+      course: gettext('Select a course or select "Not specific to a course" for your support request.'),
+      subject: gettext('Select a subject for your support request.'),
+      message: gettext('Enter some details for your support request.'),
+      request: gettext('Something went wrong. Please try again later.'),
+    };
     this.submitForm = this.submitForm.bind(this);
     this.reDirectUser = this.reDirectUser.bind(this);
-    this.setErrorState = this.setErrorState.bind(this);
     this.formOnChangeCallback = this.formOnChangeCallback.bind(this);
-    this.showWarningMessage = this.showWarningMessage.bind(this);
-    this.showDiscussionButton = this.showDiscussionButton.bind(this);
   }
 
-  setErrorState(errors) {
+  getFormDataFromState() {
+    return this.state.formData;
+  }
+
+  getFormErrorsFromState() {
+    return this.state.errorList;
+  }
+
+  clearErrorState() {
+    const formErrorsInState = this.getFormErrorsFromState();
+    Object.keys(formErrorsInState).map((index) => {
+      formErrorsInState[index] = undefined;
+      return formErrorsInState;
+    });
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  scrollToTop() {
+    return window.scrollTo(0, 0);
+  }
+
+  formHasErrors() {
+    const errorsList = this.getFormErrorsFromState();
+    return Object.keys(errorsList).filter(err => errorsList[err] !== undefined).length > 0;
+  }
+
+  updateErrorInState(key, error) {
+    const errorList = this.getFormErrorsFromState();
+    errorList[key] = error;
     this.setState({
-      errorList: errors,
+      errorList,
     });
   }
 
   formOnChangeCallback(event) {
-    const eventTarget = event.target;
-    let formData = this.state.formData;
+    const eventTarget = event.target,
```

Comment:
This seems redundant. and `event.target.value` can be directly used because of single-use.